browser/brave_ads/tabs/ads_tab_helper_browsertest.cc

```diff
@@ -0,0 +1,181 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/tabs/ads_tab_helper.h"
+
+#include <memory>
+
+#include "base/path_service.h"
+#include "brave/components/brave_ads/browser/ads_service_mock.h"
+#include "brave/components/brave_ads/core/public/prefs/pref_names.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "brave/components/constants/brave_paths.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ssl/cert_verifier_browser_test.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+// npm run test -- brave_browser_tests --filter=AdsTabHelperTest*
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kUrlDomain[] = "example.com";
+constexpr char kUrlPath[] = "/brave_ads/not_empty.html";
+constexpr char kSinglePageApplicationUrlPath[] =
+    "/brave_ads/single_page_application.html";
+
+AdsTabHelper* GetActiveAdsTabHelper(Browser* browser) {
+  auto* web_contents = browser->tab_strip_model()->GetActiveWebContents();
+  return AdsTabHelper::FromWebContents(web_contents);
+}
+
+}  // namespace
+
+class AdsTabHelperTest : public CertVerifierBrowserTest {
```

Comment:
We should have tests to check if `ads_service` is null because this is important for private browsing etc.

Reply:
fixed